kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/helpers/KmFormStateHelper.java

```diff
@@ -110,11 +112,40 @@ public static void clearInstance() {
                 for (int i = 0; i < formStateModel.getDateFieldArray().size(); i++) {
                     int key = formStateModel.getDateFieldArray().keyAt(i);
                     KmFormPayloadModel.DateTimePicker dateTimePicker = formPayloadModelList.get(key).getDatePickerModel();
-                    formDataMap.put(dateTimePicker.getLabel(), formStateModel.getDateFieldArray().get(key).toString());  //Might need to convert to formatted date
+
+                    if (KmFormPayloadModel.Type.DATE.getValue().equals(formPayloadModelList.get(key).getType())) {
+                        formDataMap.put(dateTimePicker.getLabel(), getFormattedDate(formStateModel.getDateFieldArray().get(key)));
+                    } else if (KmFormPayloadModel.Type.TIME.getValue().equals(formPayloadModelList.get(key).getType())) {
+                        formDataMap.put(dateTimePicker.getLabel(), getFormattedTime(formStateModel.getDateFieldArray().get(key)));
+                    } else {
+                        formDataMap.put(dateTimePicker.getLabel(), getFormattedDate(formStateModel.getDateFieldArray().get(key)) + "T" + getFormattedTime(formStateModel.getDateFieldArray().get(key)));
+                    }
+                }
+            }
+
+            if (formStateModel.getDropdownFieldArray() != null) {
+                for (int i = 0; i < formStateModel.getDropdownFieldArray().size(); i++) {
+                    formDataMap.put(formPayloadModelList.get(formStateModel.getDropdownFieldArray().keyAt(i)).getDropdownList().getName(), formStateModel.getDropdownFieldArray().valueAt(i).getValue());
                 }
             }
         }
 
         return formDataMap;
     }
+
+    private static String getFormattedDate(Long timeStamp) {
```

Comment:
any ready to use date formatter in existing libraries or jdk that can be used here?

Reply:
Done